sktime/forecasting/hf_momentfm_forecaster.py

```diff
@@ -0,0 +1,532 @@
+
+        # from momentfm.utils.forecasting_metrics import get_forecasting_metrics
+        from torch.optim import Adam
+        from torch.optim.lr_scheduler import OneCycleLR
+        from torch.utils.data import DataLoader
+        from tqdm import tqdm
+
+        self._pretrained_model_name_or_path = (
+            self._config["pretrained_model_name_or_path"]
+            if "pretrained_model_name_or_path" in self._config.keys()
+            else self.pretrained_model_name_or_path
+        )
+        self._freeze_encoder = (
+            self._config["freeze_encoder"]
+            if "freeze_encoder" in self._config.keys()
+            else self.freeze_encoder
+        )
+        self._freeze_embedder = (
+            self._config["freeze_embedder"]
+            if "_freeze_embedder" in self._config.keys()
+            else self.freeze_embedder
+        )
+        self._freeze_head = (
+            self._config["freeze_head"]
+            if "freeze_head" in self._config.keys()
+            else self.freeze_head
+        )
+        self._dropout = (
+            self._config["dropout"]
+            if "dropout" in self._config.keys()
+            else self.dropout
+        )
+        self._head_dropout = (
+            self._config["head_dropout"]
+            if "head_dropout" in self._config.keys()
+            else self.head_dropout
+        )
+        self._device = (
+            self._config["device"] if "device" in self._config.keys() else self.device
+        )
+        self._transformer_backbone = (
+            self._config["transformer_backbone"]
+            if "transformer_backbone" in self._config.keys()
+            else self.transformer_backbone
+        )
+        self._criterion = (
+            self._config["criterion"]
+            if "criterion" in self._config.keys()
+            else self.criterion
+        )
+        # evaluate the sequence length passed by the user
+        self._seq_len = (
+            self._config["seq_len"]
+            if "seq_len" in self._config.keys()
+            else self.seq_len
+        )
+        if self._seq_len > self._moment_seq_len:
+            warnings.warn(
+                f"length of {self._seq_len} was found which is greater than 512. "
+                "The most recent 512"
+                " values will be used when fitting.",
+                stacklevel=2,
+            )
+            self._seq_len = 512
+        # in the case the config contains 'forecasting_horizon', we'll set
+        # fh as that, otherwise we override it using the fh param
+        self._fh_config = (
+            self._config["forecasting_horizon"]
+            if "forecasting_horizon" in self._config.keys()
+            else self.fh
+        )
+        if fh is not None:
+            self._fh_input = max(fh.to_relative(self.cutoff))
+        self._fh = self._fh_input if fh is not None else self._fh_config
+        train_val_split = int(len(y) * (1 - self.train_val_split))
```

Comment:
```suggestion
        y_train, X_train, y_test, X_test = temporal_train_test_split(y, X, test_split=self.train_val_split)
```

probably needs also some updates in the code below.

Reply:
Configured the data splits to use `temporal_train_test_split`